### PR TITLE
Elastic Server Restart

### PR DESCRIPTION
**Pull request by ravindrakreddy**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ravindrakreddy** [comment]
Adding Qingqing Yuan to review the code

**Review by praries880 — CHANGES_REQUESTED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 1]
```diff
@@ -286,6 +286,41 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/restart": {
+      "post": {
+        "tags": [
+          "ServerRestart"
+        ],
+        "operationId": "Server_Restart",
```
[the remark this answers is not in the document]
Comment (ravindrakreddy) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: fixed

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json [hunk 2]
```diff
@@ -283,6 +283,41 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/restart": {
+      "post": {
+        "tags": [
+          "ServerRestart"
+        ],
+        "operationId": "Server_Restart",
```
[the remark this answers is not in the document]
Comment (ravindrakreddy) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: fixed

**Review by praries880 — APPROVED** [review]

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json [hunk 3]
```diff
@@ -283,6 +283,41 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/restart": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: "Default" response missing from all APIs. Please add. sample here - https://github.com/Azure/azure-rest-api-specs/blob/master/specification/batch/resource-manager/Microsoft.Batch/stable/2018-12-01/BatchManagement.json#L101

**Review by ravbhatnagar — COMMENTED** [review]
just one minor note. Please add the default response.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json (hunk 3 above)
Comment (ravindrakreddy) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: fixed.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Comment by praries880** [comment]
@ravindrakreddy  [Build is breaking](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/497332465#L569) because the definition of [`#/definitions/CloudError`](https://github.com/Azure/azure-rest-api-specs/pull/5244/commits/517ef876682c88a58342f72eb583818173e2a335#diff-7cd2ea152db3e6d1d7b53d7ce09d8faaR320) can not be found. Can you validate that you are referencing the correct type and its in the path specified?

**Review by praries880 — APPROVED** [review]

**Pull request merged**